mephisto/operations/worker_pool.py

```diff
@@ -265,26 +265,30 @@ async def _assign_unit_to_agent(
                     agent,
                 )
             else:
-                # See if the concurrent unit is ready to launch
                 assignment = await loop.run_in_executor(None, unit.get_assignment)
+
+                # Set status to waiting
+                agent.update_status(AgentState.STATUS_WAITING)
+
+                # See if the concurrent assignment is ready to launch
                 logger.debug(f"Attempting to launch {assignment}.")
                 agents = await loop.run_in_executor(None, assignment.get_agents)
                 if None in agents:
-                    agent.update_status(AgentState.STATUS_WAITING)
                     return  # need to wait for all agents to be here to launch
 
+                for queried_agent in agents:
+                    if queried_agent.get_status() != AgentState.STATUS_WAITING:
```

Comment:
Under what condition would this not be true? When things haven't synced to the worker threads?

Reply:
When two workers connect at the same time, and both have had an `_assign_unit_to_agent` thread called, and have thus both had `Agent`s created, but only one of those threads has made it to the `update_status` call.